src/server/routes/agencies.js

```diff
@@ -33,10 +34,19 @@ router.post('/', requireAdminUser, async function (req, res, next) {
 
   try {
     if (agencyInfo.id) {
+      const existingAgency = await agencyById(agencyInfo.id)
+      if (!existingAgency || existingAgency.tenant_id !== req.session.user.tenant_id) {
```

Comment:
huh, how would the second condition ever happen?

Reply:
Someone could do a request passing in an agency ID that is valid but refers to an agency of a different tenant.